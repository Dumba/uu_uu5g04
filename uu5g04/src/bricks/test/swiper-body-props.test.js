/**
 * Copyright (C) 2019 Unicorn a.s.
 * 
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 * 
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import React from 'react';
import {shallow} from 'enzyme';
import UU5 from "uu5g04";
import "uu5g04-bricks";
import TestTools from "../../core/test/test-tools.js";
import renderer from 'react-test-renderer';

const TagName = "UU5.Bricks.Swiper.Body";

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.PureRenderMixin"
  ],
  props: {

  },
  requiredProps: {
    parent: shallow(<UU5.Bricks.Swiper id="parentId"/>).instance()
  },
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false
    },
    enzymeToJson: false
  }
};

describe(`${TagName}`, () => {
  TestTools.testProperties(TagName, CONFIG);
});


describe(`${TagName} docKit examples`, () => {
  it(`${TagName} example01`, () => {
    const wrapper = renderer.create(
      <UU5.Bricks.Swiper
        id={"uuID"}
        style={{
          backgroundColor: "lightblue",
          minHeight: "200px",
          width: "100%"
        }}
      >
        {/*@@viewOn:0*/}
        <UU5.Bricks.Swiper.Body id={"uuID2"}>
          <UU5.Bricks.P id={"uuID3"} className="center">
            On touch screen you can open menus by swipe on blue area. If you dont have touchscreen, turn on
            console(chrome - F12) and switch to mobile device view by pressing icon at top left corner of the
            console or by pressing ctrl+shift+M.
          </UU5.Bricks.P>
        </UU5.Bricks.Swiper.Body>
        {/*@@viewOff:0*/}
      </UU5.Bricks.Swiper>
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});










