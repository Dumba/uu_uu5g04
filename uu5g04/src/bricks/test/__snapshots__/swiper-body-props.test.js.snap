// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Swiper.Body checkGrammar 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkGrammar 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkGrammar 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkGrammar 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkHighlight 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkHighlight 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkHighlight 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkHighlight 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkSpaces 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkSpaces 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkSpaces 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body checkSpaces 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body className 1`] = `
<div
  className="uu5-bricks-swiper-body my-classname"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body content 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
>
  12345
</div>
`;

exports[`UU5.Bricks.Swiper.Body content 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
>
  <UU5.Common.TextCorrectorContextConsumer
    key="0"
    text="This is Content from ContentMixin"
  />
</div>
`;

exports[`UU5.Bricks.Swiper.Body content 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
>
  <UU5.Common.TextCorrectorContextConsumer
    key="0"
    text="This is Content from ContentMixin"
  />
</div>
`;

exports[`UU5.Bricks.Swiper.Body content 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
>
  12345
</div>
`;

exports[`UU5.Bricks.Swiper.Body controlled 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body controlled 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body controlled 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body controlled 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body default props 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body disabled 1`] = `
<div
  className="uu5-bricks-swiper-body uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</div>
`;

exports[`UU5.Bricks.Swiper.Body disabled 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body disabled 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body disabled 4`] = `
<div
  className="uu5-bricks-swiper-body uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</div>
`;

exports[`UU5.Bricks.Swiper.Body docKit examples UU5.Bricks.Swiper.Body example01 1`] = `
<div
  className="uu5-bricks-swiper"
  id="uuID"
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "lightblue",
      "minHeight": "200px",
      "width": "100%",
    }
  }
  title={null}
>
  <div
    className="uu5-bricks-swiper-body"
    id="uuID2"
    style={Object {}}
    title={null}
  >
    <p
      className="uu5-bricks-p center"
      id="uuID3"
      style={Object {}}
      title={null}
    >
      On touch screen you can open menus by swipe on blue area. If you dont have touchscreen, turn on console(chrome - F12) and switch to mobile device view by pressing icon at top left corner of the console or by pressing ctrl+shift+M.
    </p>
  </div>
</div>
`;

exports[`UU5.Bricks.Swiper.Body dynamic 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body dynamic 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body dynamic 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body dynamic 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body hidden 1`] = `
<div
  className="uu5-bricks-swiper-body uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body hidden 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body hidden 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body hidden 4`] = `
<div
  className="uu5-bricks-swiper-body uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body id 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="myId"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body ignoreHTML 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body ignoreHTML 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body ignoreHTML 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body ignoreHTML 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body mainAttrs 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body name 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  name="myName"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 5`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 6`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 7`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 8`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 9`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 10`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 11`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 12`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 13`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 14`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 15`] = `""`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 16`] = `""`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 17`] = `""`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 18`] = `""`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 19`] = `""`;

exports[`UU5.Bricks.Swiper.Body nestingLevel 20`] = `""`;

exports[`UU5.Bricks.Swiper.Body noIndex 1`] = `
<div
  className="uu5-bricks-swiper-body uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body noIndex 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body noIndex 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body noIndex 4`] = `
<div
  className="uu5-bricks-swiper-body uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body pureRender 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body pureRender 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body pureRender 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body pureRender 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body selected 1`] = `
<div
  className="uu5-bricks-swiper-body uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body selected 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body selected 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body selected 4`] = `
<div
  className="uu5-bricks-swiper-body uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body style 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body style 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body style 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body style 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body textCorrector 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body textCorrector 2`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body textCorrector 3`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body textCorrector 4`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Swiper.Body tooltip 1`] = `
<div
  className="uu5-bricks-swiper-body"
  id="uuID"
  style={Object {}}
  title="My Tooltip"
/>
`;
